### DISCUSSION TITLE
Shifting windows left/right/up/down

### DISCUSSION THREAD

**Discussion by ksutrave** [rxhanson/Rectangle#521] [Ideas]
What do you think of a "shift window left/up/etc." (by one "unit", cf "make bigger/smaller") feature?

I have no idea how difficult it would be, but this is what I was hoping "move window left" did when I was looking at screenshots as I switched over from spectacle.

I also wanted to say that I'm very glad that this app exists! Thanks

**Comment by rxhanson** [comment]
Thanks for the feedback.

The main thing standing in the way of this and some other changes is that the shortcut recorder, MASShortcut, doesn't notify of up and down shortcut presses. Therefore, windows would not be continuously moved while holding the shortcut down. Without that ability I don't think I would even consider a PR for this change. So, either MASShortcut would have to be modified with that behavior (on MASShortcut's repo, with a MASShortcut release), or Rectangle would have to be refactored to swap out MASShortcut with [most likely] Sindre Sorhus' Keyboard Shortcuts recorder.

This has been asked for before, so I see the value in it and would consider a PR that includes those changes.

**Reply by ksutrave** [reply]
I can respect that, but it's strange to me. The "make bigger/smaller" feature also doesn't care about holding down the key, and I use those all the time.

**Reply by b-lov** [reply]
i'm pretty much on the same page with @ksutrave here. having the ability to move windows with the keyboard (continuously or not) would be a huge selling point for an already great app. for me personally its the last missing bit. unfortunately im not versed in macos programming at all and cant come up with a pull request. so all I can do is to beg you to reconsider your opinion about the holding-down limitation.

**Reply by rxhanson** [reply]
I don't have a desire to add this into Rectangle, but might add it into Rectangle Pro. I have a handful of window actions that I'll be adding there for an upcoming release, so I'll see about adding it at the same time.

**Reply by b-lov** [reply]
great news, i'll get the pro license as soon as this feature drops. is there any way to keep up with the latest releases of the pro version?

**Reply by rxhanson** [reply]
I'll post back here when it’s in.

**Reply by rxhanson** [reply]
Just FYI, v2.7 of [Rectangle Pro](https://rectangleapp.com/pro) has my first cut at new actions, called nudge, for moving the windows a small amount in the cardinal directions (with no continuity feature).

You can configure the amount that the windows will move with this terminal command:

defaults write com.knollsoft.Hookshot nudgeDistance -float 20

20 is the default value. You'll need to restart the app after executing that command.

**Reply by b-lov** [reply]
awesome, thanks so much 🎆

**Comment by jay-meister** [comment]
How about shifting by 1 full window width or height to the left/right or up/down respectively?

**Comment by jay-meister** [comment]
I started a new discussion: https://github.com/rxhanson/Rectangle/discussions/565
